lxd/loki/loki.go

```diff
@@ -330,14 +330,22 @@ func (c *Client) HandleEvent(event api.Event) {
 			}
 		}
 
-		messagePrefix := ""
+		keys := make([]string, len(context))
 
-		// Add the remaining context as the message prefix.
-		for k, v := range context {
-			messagePrefix += fmt.Sprintf("%s=\"%s\" ", k, v)
+		for k := range context {
+			keys = append(keys, k)
+		}
+
+		sort.Strings(keys)
+
+		var messagePrefix strings.Builder
+
+		// Add the remaining context as the message prefix. The keys are sorted alphabetically.
+		for _, k := range keys {
+			messagePrefix.WriteString(fmt.Sprintf("%s=%q ", k, context[k]))
 		}
 
-		entry.Line = fmt.Sprintf("%s%s", messagePrefix, logEvent.Message)
+		entry.Line = fmt.Sprintf("%s%s", messagePrefix.String(), logEvent.Message)
```

Comment:
You could just add `logEvent.Message` to the strings builder first and then do `entry.Line = messagePrefix.String()` here